Fixed: Product attributes labels are not translated on product edit page

fernandofauth opened the pull request:
<!--- Provide a general summary of the Pull Request in the Title above -->

The attributes on product page can't be translated.

### Description
<!--- Provide a description of the changes proposed in the pull request -->

I found where the data is provided and I simple changed the return of function add the "___()", to be can be able translation by csv file.

Page: http://prntscr.com/fswiix
Code: http://prntscr.com/fswivh
CSV: http://prntscr.com/fswjdj

### Fixed Issues (if relevant)
<!--- Provide a list of fixed issues in the format magento/magento2#<issue_number>, if relevant  -->
1. magento/magento2#1980: Issue title: Product attributes' labels are not translated on product edit page

### Manual testing scenarios
<!--- Provide a set of unambiguous steps to test the proposed code change -->
1. ...
2. ...

### Contribution checklist
 - [x] Pull request has a meaningful description of its purpose
 - [x] All commits are accompanied by meaningful commit messages
 - [x] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds on Travis CI are green)

fernandofauth committed:
Fixed bookmark translation on product list page

okorshenko commented:
Hi @fernandofauth this PR almost similar to #10184 Which one is final?

fernandofauth commented:
Ops, I'm sorry i tried splitting the commits on 2 PR.
You can considerate only this commit? "Fixed: Product attributes labels are not translated on product edit page"

okorshenko commented:
Additional information:
Fixed bookmark translation on product list page.

Old code: http://prntscr.com/fsw41t
Translation problem: http://prntscr.com/fsw5jc

New code:http://prntscr.com/fsw4w7
Translation works: http://prntscr.com/fsw5ap

fernandofauth committed:
Fixed: Product attributes labels and product attribute group label.

fernandofauth commented:
Travis rejected my changes on AbstractAttribute.php, and he's right.
In this way I removed the commit of AbstractAttribute.php .
And I found translations errors on the attribute group too.
I added all the fixes on this new commit.

fernandofauth committed:
Add translate parameter

fernandofauth committed:
Add translate parameter on labels of design_config_form.xml file

fernandofauth committed:
changed the 'echo' adding the '___()' to enable the titles translation

mageho commented:
Good afternoon,

This issue is still not fixed in Magento 2.2
Hopefully, the solution give @fernandofauth save a lot of time.

Cheers

mageho commented:
You also need to update line 581. 

Catalog Attribute Note is not translated.
Must be :
`'notice' => __($attribute->getNote()),`
